### ISSUE TITLE
Doesn't seems to create any folder / file

### ISSUE THREAD

**Issue by alanlanglois** [leethree/redux-persist-fs-storage#5]
I've successfully added the package to my expo detached project. Don't have any error, but it doesn't seems to create folder or files. 
Data aren't persisted (but are when using AsyncStorage)
Also I use redux-persist v5.

I've tried this extension: https://github.com/techwes/redux-persist-expo-fs-storage and got this [error ](https://forums.expo.io/t/expo-filesystem-redux-persist-on-detached-app/8969/)

I'm pretty new to the world of detached apps, and maybe there is something I'm missing, but can't really figured out. It's kind of critical has my users are randomly loosing data on Android using AsyncStorage...

Is there a way to check if file exists?

I'm using the pre install app "Amaze" on my genymotion emulator to check if a file exists
![image](https://user-images.githubusercontent.com/712986/38861611-450554b6-4233-11e8-9bfe-85ebdb37ecdc.png)

Thanks

**Comment by alanlanglois** [comment]
Few notes and finds while digging into the problem:
When loging store state:
   
 ```const enhancers = [applyMiddleware(...middlewares)];
    const persistConfig = { enhancers };
    const store = createStore(reducers, undefined, composeWithDevTools( ...enhancers));
    const persistor = persistStore(store, persistConfig, () => {
      console.log("fs :: getState " + JSON.stringify(store.getState()));
    });
```

Using asyncStorage, I get the complete state restitued.
When using FSStorage, I get the initial state restitued (as if nothing were saved previously. 

Also when loging if files exists, it returns true, and when I try to see if the file is correctly writed it actually is. 

When I log other methods such as "getAllKeys" or "getItem" nothing is logged as if the method were never called... Shouldn't be called when rehydrating?

**Comment by leethree** [comment]
Thanks for your report. I'll take a look.

**Comment by leethree** [comment]
@alanlanglois I tried to reproduce the issue but I couldn't. Could you post your reducer config code? 

`getItem` should be called when rehydrating as far as I know. Have you tried it on iOS or emulator other than Genymotion? 

Thanks.

**Comment by RodolfoSilva** [comment]
@alanlanglois try this:

```js
import FSStorage from 'redux-persist-fs-storage';
import RNFS from 'react-native-fs';
import { Platform } from 'react-native';

const persistDirectoryPath = Platform.select({
  ios: RNFS.LibraryDirectoryPath,
  android: RNFS.ExternalStorageDirectoryPath
});

const persistConfig = {
  key: 'store.db',
  keyPrefix: '',
  storage: FSStorage(persistDirectoryPath, 'MyApp')
};
```

**Comment by gchallen** [comment]
I just encountered the same problem. The key for me from @RodolfoSilva's answer was setting `keyPrefix: ''`. Without that keys include a colon which seems like it was producing a filename that Android could not deal with.

**Comment by leethree** [comment]
Looks like some file system doesn't allow colons. Probably I'll need to update the README to not use colon in `keyPrefix`.

**Comment by ashconnell** [comment]
PR: https://github.com/leethree/redux-persist-fs-storage/issues/5

**Comment by leethree** [comment]
Resolved by #6
